src/fu-tool.c

```diff
@@ -510,6 +534,10 @@ fu_util_get_devices (FuUtilPrivate *priv, gchar **values, GError **error)
 		}
 	}
 
+	/* save the device state for other applications to see */
```

Comment:
Wouldn't this be useful to write in the engine, so that it gets written for commands to the daemon too?

Reply:
I feel like `state.json` as a `fwupdtool` specific _hac^b_ behavior.  If you have the daemon why would you use it?  You should just contact the daemon over dbus to get all this information.

The only situation I can think that it makes sense on a system with a daemon is to analyze after the daemon has automatically shut down (like a timeout).

Reply:
>  You should just contact the daemon over dbus to get all this information

True...